### PR TITLE
Adding multiple layers hide show to Layer->Visible (shift-x)

### PR DESCRIPTION
**Pull request by robotron2084**
This adds multiple layer support to Shift-X and Layer->Visible via LayerVisibilityCommand. I did not modify the layer palette to change multiple layers, but I can def add that to this review if you request.

Additionally I forked the repo and did various git incantations then forgot to branch *before* committing my change (i.e. I committed to master then branched). The PR looks OK and hopefully git is forgiving of this.

### CODE REVIEW

File: src/app/commands/cmd_layer_visibility.cpp [hunk 1]
```diff
@@ -45,16 +45,31 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
 bool LayerVisibilityCommand::onChecked(Context* context)
 {
   const ContextReader reader(context);
-  const Layer* layer = reader.layer();
-  return (layer && layer->isVisible());
+  const SelectedLayers& selLayers = reader.site()->selectedLayers();
+  bool visible = false;
+  for (auto layer : selLayers)
+  {
+    if(layer && layer->isVisible())
```
Comment (dacap) [inline src/app/commands/cmd_layer_visibility.cpp]: Minor style change:
```for (auto layer : selLayers) {
  if (layer && layer->isVisible())
```

File: src/app/commands/cmd_layer_visibility.cpp [hunk 2]
```diff
@@ -45,16 +45,31 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
...
 void LayerVisibilityCommand::onExecute(Context* context)
 {
   ContextWriter writer(context);
-  Layer* layer = writer.layer();
+  const SelectedLayers& selLayers = writer.site()->selectedLayers();
 
-  layer->setVisible(!layer->isVisible());
+  bool anyVisible = false;
+  for (auto layer : selLayers)
+  {
```
Comment (dacap) [inline src/app/commands/cmd_layer_visibility.cpp]: Same style that above.

File: src/app/commands/cmd_layer_visibility.cpp [hunk 3]
```diff
@@ -45,16 +45,31 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
...
+      visible = true;
+  }
+  return visible;
 }
 
 void LayerVisibilityCommand::onExecute(Context* context)
 {
   ContextWriter writer(context);
-  Layer* layer = writer.layer();
+  const SelectedLayers& selLayers = writer.site()->selectedLayers();
```
Comment (dacap) [inline src/app/commands/cmd_layer_visibility.cpp]: Actually now the command doesn't work for the single layer case. This is the case when `selLayers` is empty and `writer.layer()` is the current layer (this happens when we select a layer and then click the sprite editor, the selection outline in the timeline disappears).

You can use something like this here (and in `onChecked`):

    SelectedLayers selLayers;
    auto range = App::instance()->timeline()->range();
    if (range.enabled()) {
      selLayers = range.selectedLayers();
    }
    else {
      selLayers.insert(writer.layer());
    }

Comment (robotron2084) [inline src/app/commands/cmd_layer_visibility.cpp]: oh wow, I can't believe I didn't test that use case. I will code something similar and update the PR.

File: src/app/commands/cmd_layer_visibility.cpp (hunk 1 above)
Comment (robotron2084) [inline src/app/commands/cmd_layer_visibility.cpp]: Got it! I will do my best to follow the style guide.

File: src/app/commands/cmd_layer_visibility.cpp (hunk 3 above)
Comment (dacap) [inline src/app/commands/cmd_layer_visibility.cpp]: No problem. I should add a function to get the real set of selected layers and avoid this kind of problems in the future and simplify code (also this little code is duplicated in several places, the `writer.site()` should have a function to solve this for us already).

Comment (robotron2084) [inline src/app/commands/cmd_layer_visibility.cpp]: If that is the better place for this to exist then perhaps I can do that instead. My only thought about that is how best to handle the memory .

Comment (dacap) [inline src/app/commands/cmd_layer_visibility.cpp]: Don't worry, you can now copy the lines, I'll see how to do it right (or if you want, you can then create then another PR for that refactor).

File: src/app/commands/cmd_layer_visibility.cpp [hunk 4]
```diff
@@ -45,10 +47,16 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
...
 {
   const ContextReader reader(context);
-  const SelectedLayers& selLayers = reader.site()->selectedLayers();
+  SelectedLayers selLayers;
+  auto range = App::instance()->timeline()->range();
+  if (range.enabled()) {
+    selLayers = range.selectedLayers();
+  }
+  else {
+    selLayers.insert(const_cast<Layer*>(reader.layer()));
```
Comment (robotron2084) [inline src/app/commands/cmd_layer_visibility.cpp]: I felt a bit dirty const_casting this. I looked for examples of usage with reader.layer() in other commands following the pattern you suggested and didn't find any. I can modify this code to use logic that doesn't cast away the const if you like.

**Comment by dacap** [comment]
This PR was merged with commits squashed (d0c1be871a413f649f11c33549b9984f02fb6ea6). Thanks Chris!

**Pull request closed without merging**